browser/ts/index.ts

```diff
@@ -199,6 +201,9 @@ export interface Config {
     // all the required fields) are automatically included. For more information
     // on JSON Schemas, see https://json-schema.org/
     customOrderFilter?: JsonSchema;
+    // Offers the ability to use your own web3 provider for all Ethereum RPC
+    // requests instead of the default.
+    web3Provider?: Provider;
```

Comment:
@fabioberger Is `Provider ` the right type to use here or should we be accepting a more general type like `SupportedProvider` or `ZeroExProvider`? The main thing to keep in mind is that we need a consistent interface that our Go code can call out to. Maybe there's a way to convert to a standard interface?

Reply:
Can we set it to `SupportedProvider` and then call `@0x/utils`'s [`providerUtils.standardizeOrThrow()`](https://github.com/0xProject/0x-monorepo/blob/development/packages/utils/src/provider_utils.ts#L33) method? That'll spit out a `ZeroExProvider` that we can rely on to always have a consistent interface. Unfortunately there was never a formal standard for what a "provider" looks like and so there are _many_ implementations floating around.